File: Java/sga/model/ProgramaAcademico.java
package sga.model;

import java.util.List;

public abstract class ProgramaAcademico {
    protected String nombrePrograma;

    public ProgramaAcademico(String nombrePrograma) {
        this.nombrePrograma = nombrePrograma;
    }

    public String getNombrePrograma() {
        return nombrePrograma;
    }

    public abstract boolean evaluarAprobacion(List<Float> notas);
}

File: C++/ProgramaAcademico.cpp
#include "ProgramaAcademico.h"

ProgramaAcademico::ProgramaAcademico(const std::string& nombrePrograma)
    : nombrePrograma(nombrePrograma) {}

std::string ProgramaAcademico::getNombrePrograma() const {
    return nombrePrograma;
}